### PR TITLE
Add link to notifications which links to the corresponding posts.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/components/Notification/index.jsx [hunk 1]
```diff
@@ -54,7 +56,7 @@ let NotifiItem = React.createClass({
                         <div className="name" style={nameStyle}><Name member={this.props.sender}/></div>
                         <div className="time" style={timeStyle} flex={1}><SmartTimeDisplay start={this.props.time} relative/></div>
                     </Layout>
-                    <div style={messageStyle}>{this.props.message}</div>
+                    <a href={this.props.url} style={messageStyle}>{this.props.message}</a>
```
Comment (LeonLiuY) [inline app/components/Notification/index.jsx]: `<a>` will cause the whole page refresh, which is not good UX.

Consider to use react router and do transition without refreshing the whole page.

Comment (LeonLiuY) [inline app/components/Notification/index.jsx]:
```
contextTypes: {
    router: React.PropTypes.func.isRequired
},
...
this.context.router.transitionTo(...)
```

File: app/components/Thread/index.jsx [hunk 2]
```diff
@@ -141,7 +141,7 @@ const Thread = React.createClass({
                     email_subject: `[RhinoBird] ${LoginStore.getUser().realname} mentioned you in comment`,
                     email_body: `${LoginStore.getUser().realname} mentioned you in comment of <a href="${window.location.href.split('#')[0]}">${this.props.threadTitle}</a>
                     <br/><br/> ${comment}`,
-                    url: `window.location.pathname#${c.cid}`
+                    url: window.location.href.split('#')[0]+"#" + c.cid
```
Comment (LeonLiuY) [inline app/components/Thread/index.jsx]: What's the reason to change this line?

Comment (ghost) [inline app/components/Thread/index.jsx]: The previous line `window.location.pathname#${c.cid}` dosen't give the right url. The variable window.location.pathname isn't populated with a true value, but treated as a plain text string.

Comment (LeonLiuY) [inline app/components/Thread/index.jsx]: I got, it is small mistake. BTW, you can change it to

```
`${window.location.pathname}#${c.cid}`
```

I think it is better than `href.split`

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/template_strings

File: app/components/Notification/index.jsx [hunk 3]
```diff
@@ -21,17 +21,26 @@ let NotifiItem = React.createClass({
...
     contextTypes: {
-        muiTheme: React.PropTypes.object
+        muiTheme: React.PropTypes.object,
+        router: React.PropTypes.func.isRequired
+    },
+
+    _transitionToPost(){
+        var parser = document.createElement('a');
+        parser.href = this.props.url;
+        this.context.router.transitionTo(parser.pathname);
```
Comment (LeonLiuY) [inline app/components/Notification/index.jsx]: What's the problem of just using `this.context.router.transitionTo(this.props.url);` ?

File: app/components/Notification/index.jsx [hunk 4]
```diff
@@ -54,7 +66,7 @@ let NotifiItem = React.createClass({
                         <div className="name" style={nameStyle}><Name member={this.props.sender}/></div>
                         <div className="time" style={timeStyle} flex={1}><SmartTimeDisplay start={this.props.time} relative/></div>
                     </Layout>
-                    <div style={messageStyle}>{this.props.message}</div>
+                    <span style={achorStyle} onClick={this._transitionToPost}>{this.props.message}</span>
```
Comment (LeonLiuY) [inline app/components/Notification/index.jsx]: FYI, you can use https://github.com/rhinobird-io/rhinobird-web/blob/master/app/components/Common/RouterLink.jsx

Comment (LeonLiuY) [inline app/components/Notification/index.jsx]: http://rackt.github.io/react-router/#Link

**Pull request merged**